sklearn/tests/metadata_routing_common.py

```diff
@@ -53,27 +61,34 @@ def check_recorded_metadata(obj, method, split_params=tuple(), **kwargs):
         sub-estimator to check routed params for
     method : str
         sub-estimator's method where metadata is routed to
+    parent : str
+        the parent method which should have called `method` or caller
```

Comment:
Just to add to the discussion: having a parent method here would help simplify the unit-testing in https://github.com/scikit-learn/scikit-learn/pull/28494/files#diff-fb0682201360bf8493836d8fb3977b58e035dd084e07dec9a8522f64df94bb38?